### ISSUE TITLE
Trade.equals()

### ISSUE THREAD

**Issue by bryan4887** [timmolter/XChange#808]
This seemed like a sufficiently distinct issue to https://github.com/timmolter/XChange/pull/219, so I chose to open a new one.

This is the current `Trade.equals()`:
```
  public boolean equals(Object o) {

    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    return this.id.equals(((Trade) o).getId());
  }
```
I think the `return` line is naive.  What if somehow a Trade gets created with different fields but equivalent ids?  Obviously this doesn't make sense, but the current implementation wouldn't catch it either.  

I am in favor of defining equality by calling `Object.equals()` or `==` on each of the two Trades' fields.  Additionally an exception could be thrown to catch weird scenarios like equivalent ids but different field values.

Basically, I'm in favor of the most conservative definition of `equals`.
Thoughts?

**Comment by timmolter** [comment]
@bryan4887 I cannot image a two different trades with the same ID coming from an exchange. Have you seen this before?

**Comment by rafalkrupinski** [comment]
Trade expected = new Trade(/*values*/); // no id
    expected.equals(incoming) // always false

**Comment by timmolter** [comment]
@rafalkrupinski And why would it have no id? I think the assumption has to be made that all trades have an id. And then comparing other fields also wouldn't work either. Different trades could have the same amount/price. and timestamp.

**Comment by rafalkrupinski** [comment]
@timmolter OK, you're right.

**Comment by rafalkrupinski** [comment]
I think getting UserTrade and Trade with the same id is possible, so in this case the

    if (getClass() != o.getClass()) {
      return false;
    }

part is a bug.
